HW#3 - 1st commit

drag0nfet opened the pull request:
- реализовал все пункты домашки, только эксепшены сразу перехватываются в класе Gender

drag0nfet committed:
HW#3 - 1st commit
- реализовал все пункты домашки, только эксепшены сразу перехватываются в класе Gender

Diagorn commented on ru/turing/courses/lesson4/Gender.java:
```diff
@@ -24,12 +24,44 @@ public enum Gender {
+                }
+            }
+            throw new GenderNotFoundException("Not found gender with name " + name);
+        } catch (GenderNotFoundException e){
```
Зачем ловить это тут? Тогда смысла в эксепшне нет вообще. Брось это всё в вызывающий код

Diagorn commented on ru/turing/courses/lesson4/Gender.java:
```diff
@@ -24,12 +24,44 @@ public enum Gender {
     public static Gender ofName(String name) {
-        //todo реализовать с кастомным эксепшном
+        try {
+            if (name == null)
```
Даже если внутри `if() {}` всего одна строка, фигурные скобки нужно ставить

Diagorn commented on ru/turing/courses/lesson4/Gender.java:
```diff
@@ -24,12 +24,44 @@ public enum Gender {
+            if (name == null)
+                throw new GenderNotFoundException("Null name!");
+
+            String newName = name.toLowerCase();
```
Лишняя переменная, можно сравнивать прямо в условии

Diagorn commented on ru/turing/courses/lesson4/Gender.java:
```diff
@@ -24,12 +24,44 @@ public enum Gender {
+                throw new GenderNotFoundException("Null name!");
+
+            String newName = name.toLowerCase();
+            String myName;
```
Название переменной не самое удачное + переменную можно вытащить в цикл

Diagorn commented on ru/turing/courses/lesson4/Gender.java:
```diff
@@ -24,12 +24,44 @@ public enum Gender {
         return null;
     }
 
     public static Gender ofShortNameRepr(Character shortNameRepr) {
```
Все замечания из предыдущего метода справедливы и здесь

Diagorn commented on ru/turing/courses/lesson4/Main.java:
```diff
@@ -3,9 +3,9 @@
     public static void main(String[] args) {
 
-        //todo реализовать с учетом выбросов эксепшнов следующие кейсы
+        // ��������� ���������� ����� � ������ Gender, ����� �� ������� User Interface
```
Слетела кодировка в комментах

drag0nfet committed:
HW#3 - 2nd fix commit
- эксепшены перехватываются в main, убрал лишние переменные, поправил форматирование

Diagorn commented on ru/turing/courses/lesson4/Gender.java:
```diff
@@ -23,14 +29,33 @@ public enum Gender {
+        }
+
+        for (Gender obj : values()) {
+            if ((obj.shortNameRepr + "").equalsIgnoreCase(shortNameRepr + "")) {
```
`obj.shortNameRepr + ""`
Это ещё зачем? У тебя `shortNameRepr` - примитив, он через `==` спокойно работает, накрайняк - `Objects.equals()`

drag0nfet committed:
HW#3 - 3rd fix commit
- изменил в 53 строке класса Gender сравнение по короткому обозначению (изначально задумка была в том, чтобы от регистра не зависело, теперь, например, по краткому обозначению F будет выкидывать exception. Но по факту работать, конечно, будет)

Diagorn reviewed (approved):
Молодец!
